### PR TITLE
fix: better handling of text responses

### PR DESCRIPTION
**Pull request by sg-s**
## changes

small fix to handle text responses

### CODE REVIEW

File: src/data_hub/_api.py [hunk 1]
```diff
@@ -183,12 +184,17 @@ def dynamic_function(
             )
 
         if not isinstance(response, dict):
-            response = response.json()
+            try:
+                response = response.json()
+            except JSONDecodeError:
+                response = dict(data=response.text())
```
Comment (kennyjwilli) [inline src/data_hub/_api.py:190]: Fine for now, but a better solution would be to dispatch on the response `content-type` header.

**Review by kennyjwilli — APPROVED** [review]

File: src/data_hub/_api.py (hunk 1 above)
Comment (sg-s) [reply src/data_hub/_api.py:190]: i should check for `'text/event-stream'`?

Comment (sg-s) [reply src/data_hub/_api.py:190]: looks like it's either `application/json` or `text/event-stream`?

Comment (sg-s) [reply src/data_hub/_api.py]: fixed

**Pull request merged**